src/js/base/fetch.js

```diff
@@ -0,0 +1,64 @@
+//  Similar to https://github.com/akre54/Backbone.Fetch
+import $ from 'jquery';
+import { isObject, defaults, extend } from 'underscore';
+import Radio from 'backbone.radio';
+
+import { getToken } from 'js/auth';
+
+// Makes data object into `/url?param1=value1&param2=value2` string
+function getUrl(url, { data, method }) {
+  if (method !== 'GET' || !isObject(data)) return url;
+
+  const params = $.param(data);
+  if (!params) return url;
+
+  return `${ url }?${ params }`;
+}
+
+export function getData(response, dataType) {
+  if (dataType === 'json' && response.status !== 204) return response.json();
+  return response.text();
+}
+
+export function handleJSON(response) {
+  if (!response.ok) return Promise.reject(response);
+
+  return getData(response, 'json');
+}
+
+export default async(url, opts) => {
```

Comment:
I haven't had extensive experience with `fetch`, but this brings fetch to general parity with ajax with an attached `abort` and sensible defaults for our app along with the global 401/5xx handlers.